### PR TITLE
updated resolver to be reversible, added bencat's nolerp

### PR DESCRIPTION
**Pull request by dotqurter**
resolver works well enough, i should probably revert the nolerp settings, as it seemed to cause a massive frame drop but that may have just been a bug

### CODE REVIEW

**Comment by TotallyNotElite** [comment]
Can you explain what this pull request tries to accomplish? What issue does this solve? Do you have an issue #?

**Comment by dotqurter** [comment]
it fixes the resolver to use a more-proper bruteforce "algorithm" (if you wanna call it that)

**Comment by TotallyNotElite** [comment]
There is definitely a lot more than that. Can you summarize the full functionality of this patch?

File: src/hacks/AntiAntiAim.cpp [hunk 1]
```diff
@@ -27,6 +27,19 @@ static inline void modifyAnlges()
...
+
+static inline void resetangles()
+{
+    for (int i = 1; i <= g_IEngine->GetMaxClients(); i++)
+    {
+        auto player = ENTITY(i);
+        if (CE_BAD(player) || !player->m_bAlivePlayer() || !player->m_bEnemy() || !player->player_info.friendsID)
+            continue;
+        auto &data  = resolver_map[player->player_info.friendsID];
+        CE_VECTOR(player, netvar.m_angEyeAngles) = CE_VECTOR(player, netvar.m_angEyeAngles);
```
Comment (BenCat07) [inline src/hacks/AntiAntiAim.cpp:39]: set to itself?

Comment (dotqurter) [reply src/hacks/AntiAntiAim.cpp:39]: yeah, it was the way i fixed the issue where it wouldn't disable after turning off

**Comment by dotqurter** [comment]
> There is definitely a lot more than that. Can you summarize the full functionality of this patch?

there isn't much to it: the resolver part just makes a line from localplayer to enemyplayer, and uses bruteforce to tell which perpendicular angle to move the head to; so left or right

**Comment by TotallyNotElite** [comment]
Oh I thought "resolver to be reversible" implied some new feature instead of improvements to the existing one. I'll take a look later.

**Comment by dotqurter** [comment]
no, its a semi-joke because the last one that I had refused to disable if you turned off the feature, so you couldn't disable the resolver; this one actually works

**Comment by TotallyNotElite** [comment]
Going to leave this review to @BenCat07 because he knows more about antiantiaim than me

File: src/hacks/AntiAntiAim.cpp [hunk 2]
```diff
@@ -27,6 +27,19 @@ static inline void modifyAnlges()
...
 }
+
+static inline void resetangles()
+{
+    for (int i = 1; i <= g_IEngine->GetMaxClients(); i++)
+    {
+        auto player = ENTITY(i);
+        if (CE_BAD(player) || !player->m_bAlivePlayer() || !player->m_bEnemy() || !player->player_info.friendsID)
+            continue;
+        auto &data  = resolver_map[player->player_info.friendsID];
```
Comment (BenCat07) [inline src/hacks/AntiAntiAim.cpp:38]: This is unused

File: src/hacks/AntiAntiAim.cpp [hunk 3]
```diff
@@ -112,14 +149,22 @@ static float resolveAnglePitch(float angle, brutedata &brute, CachedEntity *ent)
...
+        if (brute.brutenum % 3)
         {
             // Pitch resolver
             if (angle >= 90)
                 angle = -89;
-            if (angle <= -90)
+            else if (angle <= -90)
                 angle = 89;
+            else
+                angle = 0;
```
Comment (BenCat07) [inline src/hacks/AntiAntiAim.cpp:161]: This change looks good actually

File: src/hooks/visual/FrameStageNotify.cpp [hunk 4]
```diff
@@ -18,6 +18,7 @@ static settings::Rgba nightmode_gui_color{ "visual.night-mode.gui-color", "00000
 static settings::Rgba nightmode_world_color{ "visual.night-mode.world-color", "000000FF" };
 static settings::Rgba nightmode_skybox_color{ "visual.night-mode.skybox-color", "000000FF" };
 static settings::Boolean no_shake{ "visual.no-shake", "true" };
+static settings::Boolean nointerp_enabled("misc.animfix.nointerp", "false");
```
Comment (BenCat07) [inline src/hooks/visual/FrameStageNotify.cpp:21]: We should combine all of these including the original "animfix" into the nolerp variable, could you maybe do that for me? thanks

**Review by BenCat07 — CHANGES_REQUESTED** [review]

**Comment by BenCat07** [comment]
@dotqurter fyi i reviewed

**Comment by incontestableness** [comment]
@dotqurter If you don't finish this I might get around to it eventually
@BenCat07 what's the nolerp fix do exactly? current nolerp sucks

**Comment by TotallyNotElite** [comment]
Closing due to inactivity, feel free to reopen if interested.
